File: src/app/models/AccountStatement.js

// Define collection and schema for AccountStatement
export interface AccountStatement {
    statementNumber:
	type : string
    periodStart:
	type : Date
    periodEnd:
	type : Date
    openingBalance:
	type : Money
    closingBalance:
	type : Money
    Account:
	type : Schema.Types.ObjectId
    DeliveryMethod:
 	type : String
#
    collection: 'accountStatements'
}

File: bankingOnGolang/internal/model/AccountStatement.go
package model

import (
	"time"
)

// ==============================================================
// AccountStatement Declaration
// ==============================================================
type AccountStatement struct {
	BaseModel
	StatementNumber string
	PeriodStart     time.Time
	PeriodEnd       time.Time
	OpeningBalance  Money
	ClosingBalance  Money
	AccountId       *uint
	Account         *Account `gorm:"foreignKey:AccountId"`
	DeliveryMethod  StatementDeliveryMethod

	// parent associations as their child

}
